### ISSUE TITLE
API Proposal: Broadcasts with Layout

### ISSUE THREAD

**Issue by aoberoi** [opentok/Opentok-.NET-SDK#67]
This is a request for comments regarding the API for upcoming platform functionality. The platform functionality is not generally available at the moment and there is no guarantee it will be released in any form.

this is related to #66 
### Create a Broadcast

``` csharp
// each returns an instance of `Broadcast`

// simplest case (`sessionId` is a String)
var broadcast = opentok.StartBroadcast(sessionId);

// using predefined layout (`Layout.PIP` is an instance of `Layout` and a static property of the `Layout` class)
var broadcast = opentok.StartBroadcast(sessionId, layout=Layout.PIP);

// using a custom layout (`Layout.CreateCustom()` returns an instance of `Layout`, `stylesheet` is a CSS string)
var broadcast = opentok.StartBroadcast(sessionId, layout=Layout.CreateCustom(stylesheet=stylesheet));
```
### Stop a Broadcast

``` csharp
// when you don't have a `Broadcast` instance (`broadcastId` is a String)
var broadcast = opentok.StopBroadcast(broadcastId);

broadcast.Stop();
```
### Get a Broadcast

``` csharp
// `broadcastId` is a String
var broadcast = opentok.GetBroadcast(broadcastId);

// `Broadcast` has the following properties
broadcast.Id; // string
broadcast.SessionId; // string
broadcast.CreatedAt; // long
broadcast.UpdatedAt; // long
broadcast.PartnerId; // string
broadcast.Urls; // Dictionary<string, string> contains a key `"HLS"` with a value of a URL
```
### Get a Broadcast Layout

``` csharp
// each returns a Layout instance

// when you don't have a `Broadcast` instance (`broadcastId` is a String)
var layout = opentok.GetBroadcastLayout(broadcastId);

var layout = broadcast.GetLayout()
```
### Update a Broadcast Layout

``` csharp
// when you don't have a `Broadcast` instance (`broadcastId` is a String, `layout` is an instance of `Layout`)
opentok.UpdateBroadcastLayout(broadcastId, layout);

// `layout` is an instance of `Layout`
broadcast.UpdateLayout(layout);
```
### Update a Stream's Layout Class List

Question: should we have an API for updating multiple streams?
Question: should these return a representation of the Stream properties?

``` csharp
// when you don't have a `Session` instance
opentok.UpdateStream(sessionId, streamId, layoutClassList=new string[] { "class1" });

session.UpdateStream(streamId, layoutClassList=new string[] { "class1", "class2" });
```

**Comment by aoberoi** [comment]
we plan to begin an implementation next week. please comment at your earliest convenience.

**Comment by msach22** [comment]
This has been released with `v3.1.0`.
